common/settings/settings.ts

```diff
@@ -30,6 +30,86 @@ export interface MiscSettings {
     readonly pauseOnHoverMode: PauseOnHoverMode;
 }
 
+/*
+These are all the possible scenarios which can result in a match. We don't need to support every possible combination,
+as some are not useful or inconsistent. Inconsistent meaning the order the user collects forms affects what future forms
+are considered collected (e.g collecting the lemma will match all forms but user needs to collect every inflection if
+they don't ever collect the lemma).
+
+Lemma In Subtitle
+-----------------------------------------------------------------
+| User Collection | LEMMA_FORM_COLLECTED | EXACT_FORM_COLLECTED |
+-----------------------------------------------------------------
+| Lemma           |         MATCH        |         MATCH        |
+| Inflection      |          NO          |          NO          |
+-----------------------------------------------------------------
+
+Inflection In Subtitle
+-----------------------------------------------------------------
+| User Collection | LEMMA_FORM_COLLECTED | EXACT_FORM_COLLECTED |
+-----------------------------------------------------------------
+| Lemma           |         MATCH        |          NO          |
+| Same Inflection |          NO          |         MATCH        |
+| Diff Inflection |          NO          |          NO          |
+-----------------------------------------------------------------
+*/
+export enum TokenMatchStrategy {
+    ANY_FORM_COLLECTED = 'ANY_FORM_COLLECTED', // All scenarios above result in MATCH
+    LEMMA_OR_EXACT_FORM_COLLECTED = 'LEMMA_OR_EXACT_FORM_COLLECTED', // See LEMMA_FORM_COLLECTED and EXACT_FORM_COLLECTED columns above
+    LEMMA_FORM_COLLECTED = 'LEMMA_FORM_COLLECTED', // See LEMMA_FORM_COLLECTED column above
+    EXACT_FORM_COLLECTED = 'EXACT_FORM_COLLECTED', // See EXACT_FORM_COLLECTED column above
+}
+
+export enum TokenMatchStrategyPriority {
+    EXACT = 'EXACT',
```

Comment:
Something I noticed but I'm not sure if it's a problem. EXACT prioritized lemma form in the word field ahead of exact form in the sentence field, so it feels like a misnomer. Is this because word fields are more important than sentence fields like you mentioned before?